### DISCUSSION TITLE
Marp Cli Engine (--engine) with the API (node.js)

### DISCUSSION THREAD

**Discussion by samuele-cozzi** [marp-team/marp#488] [Q&A] [answered]
Hi, I appreciate your wonderful repository. 
Does the engine parameter (--engine) get supported by the [API (Experimental)](https://github.com/marp-team/marp-cli?tab=readme-ov-file#api-experimental)? Exist any practical examples?

**Comment by yhatt** [comment, marked as answer]
Marp CLI API supports any CLI arguments including `--engine`.  (In other words, API does never restrict any CLI options)

```javascript
import { marpCli } from '@marp-team/marp-cli'

// Specify engine by the module name
marpCli(['--engine', '@marp-team/marp-core', './slide.md'])

// Specify engine by the path for the function engine
marpCli(['--engine', 'engine.mjs', './slide.md'])
```

**Reply by samuele-cozzi** [reply]
Hi,

I'm developing [obsidian plugin](https://github.com/samuele-cozzi/obsidian-marp-slides) on electron framework. I have tried the first one but without success: 
![image](https://github.com/marp-team/marp/assets/2479040/8e901085-1c26-4ad7-a171-64709717c501)

I'm trying the second one but it doesn't work.

Do you have some working examples of this implementation, or do you have any suggestion on this case?

**Reply by yhatt** [reply]
If specified the module name as `--engine` option, Marp CLI tries to find out the specified module from `node_modules` in the current Node.js project. Within Electron-based apps such as Obsidian, it may not work because the project struture is different from general Node.js projects.

So we recommend to use the external script as `--engine` option in your use case. Actually [an another Marp plugin for Obsidian is using this way](https://github.com/JichouP/obsidian-marp-plugin/blob/23eaf74da54ce2160b0ef7869fd1f139d1259397/src/export.ts#L48) and working well. `engine.mjs` may be bundled while building the plugin, so it's important for using the custom engine to keep a splitted Marp engine script file.

**Reply by ruthkangyr** [reply]
Hi,

Im trying to use the marpCli in my nextjs webpage to obtain the output (pdf or pptx). I did this but it does not work.

```
const tempMarkdownPath = join(tmpdir(), ${uuidv4()}.md)
await writeFile(tempMarkdownPath, markdownContent)

 
const tempPdfPath = join(tmpdir(), ${uuidv4()}.pdf)
const exitStatus = await marpCli([tempMarkdownPath, '--pdf', '--output', tempPdfPath])
```

Can i ask if you have some working examples of this implementation, or do you have any suggestion on this case? Thank you
